sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/component.py

```diff
@@ -208,7 +208,7 @@ def _get_snapshot_id(cls, code_path: Union[str, PathLike]) -> str:
         :type code_path: str
         :return: The snapshot id of a component in ml-components with code_path as its working directory.
         """
-        _ignore_file: IgnoreFile = get_ignore_file(code_path)
+        _ignore_file = InternalComponentIgnoreFile.from_ignore_file(get_ignore_file(code_path))
```

Comment:
maybe make ignore file a property of InternalComponent?

Reply:
sure

Reply:
please check the newest code